src/view.c

```diff
@@ -311,8 +311,10 @@ void window_node_flush(struct window_node *node)
             if (window) {
                 if (node->zoom) {
                     window_manager_set_window_frame(window, node->zoom->area.x, node->zoom->area.y, node->zoom->area.w, node->zoom->area.h);
+                    border_resize(window);
                 } else {
                     window_manager_set_window_frame(window, node->area.x, node->area.y, node->area.w, node->area.h);
+                    border_resize(window);
```

Comment:
these added lines are the only call from `view.c` into `border.{h,c}`, so I suspect that it may violate an encapsulation concept you are going for - and that these calls belong inside `window_manager_set_window_frame` or indeed in `window_manager_resize_window` - but those functions confuse me more (with the AX_ENHANCED_UI_WORKAROUND stuff), so I shoved in here until told otherwise

Reply:
I also felt like these calls should be conditionalized on whether borders are on, but since the first thing resize does is return if there is no border, seemed fine?